src/plugins/intel_gpu/src/graph/CMakeLists.txt

```
@@ -4,76 +4,118 @@
-    list(REMOVE_ITEM LIBRARY_SRC ${SYCL_SOURCES})
-endif()
-
-add_library(${TARGET_NAME} STATIC ${LIBRARY_SRC})
-
-target_include_directories(${TARGET_NAME} PUBLIC
-                                          $<BUILD_INTERFACE:${MAIN_DIR}/src>
-                                          $<BUILD_INTERFACE:${INCLUDE_DIR}>
-                                          $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-                                          $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/>)
-
-target_compile_options(${TARGET_NAME} PRIVATE
-  $<$<CONFIG:Release>:$<IF:$<CXX_COMPILER_ID:MSVC>,/Os,-Os>>)
-
-if(OV_COMPILER_IS_INTEL_LLVM)
-    add_sycl_to_target(TARGET ${TARGET_NAME} SOURCES ${SYCL_SOURCES})
-    target_compile_definitions(${TARGET_NAME} PUBLIC OV_GPU_WITH_SYCL)
-endif()
-
-target_link_libraries(${TARGET_NAME} PUBLIC OpenCL::OpenCL openvino::shape_inference)
-target_link_libraries(${TARGET_NAME} PRIVATE openvino_intel_gpu_kernels
-                                             openvino_intel_gpu_runtime
-                                             openvino::itt
-                                             openvino::reference
-                                             openvino::runtime::dev
-                                             openvino::runtime)
-
-if(ENABLE_ONEDNN_FOR_GPU)
-    ov_target_link_libraries_as_system(${TARGET_NAME} PUBLIC onednn_gpu_tgt)
-endif()
-
-if(COMMAND add_cpplint_target)
-  add_cpplint_target(${TARGET_NAME}_cpplint FOR_TARGETS ${TARGET_NAME})
-endif()
+file(GLOB SUBDIRS "${CMAKE_CURRENT_SOURCE_DIR}/impls/*")
+
+foreach(SUBDIR IN LISTS SUBDIRS)
+    if(IS_DIRECTORY ${SUBDIR})
+        get_filename_component(SUBDIR_NAME ${SUBDIR} NAME)
+        if (NOT SUBDIR_NAME STREQUAL "registry")
+            list(APPEND AVAILABLE_IMPL_TYPES ${SUBDIR_NAME})
+            list(APPEND EXCLUDE_PATHS ${SUBDIR})
+        endif()
+    endif()
+endforeach()
+
+set(COMMON_COMPILE_OPTIONS $<$<CONFIG:Release>:$<IF:$<CXX_COMPILER_ID:MSVC>,/Os,-Os -Wno-suggest-override>>)
+set(COMMON_INCLUDE_DIRS $<BUILD_INTERFACE:${MAIN_DIR}/src>
+                        $<BUILD_INTERFACE:${INCLUDE_DIR}>
+                        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+                        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+)
+
+set(COMMON_LINK_LIBRARIES openvino::shape_inference # for tensor accessor
+                          openvino_intel_gpu_kernels # for cl_kernel_data_serializer
+                          openvino_intel_gpu_runtime
+                          openvino::itt
+                          openvino::runtime::dev
+                          openvino::runtime
+)
+
+define_property(GLOBAL PROPERTY EXTRA_LINK_LIBRARIES_GLOBAL BRIEF_DOCS "All link libs" FULL_DOCS "Link libraries collection from all backends")
+set_property(GLOBAL PROPERTY EXTRA_LINK_LIBRARIES_GLOBAL "")
+
+function(ov_gpu_add_backend_target)
+    set(options
+    )
+    set(oneValueRequiredArgs
+        NAME
```

```suggestion
        NAME
        ROOT
```
if you still want to keep `ROOT`